modules/miner/miner.go

```diff
@@ -194,10 +199,15 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
```

Comment:
Hmm. Alternatively you might want to just check the value of `tg.IsStopped()` inside of the `threadedMining` thread.